apps/explorer/src/hooks/useSearch.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    isValidTransactionDigest,
+    type SuiTransactionResponse,
+    type JsonRpcProvider,
+    isValidSuiAddress,
+    isValidSuiObjectId,
+    normalizeSuiObjectId,
+    type GetObjectDataResponse,
+    is,
+    SuiObject,
+} from '@mysten/sui.js';
+import { useQuery } from '@tanstack/react-query';
+
+import { useRpc } from '~/hooks/useRpc';
+import { isGenesisLibAddress } from '~/utils/api/searchUtil';
+
+const handleSearch = async (rpc: JsonRpcProvider, query: string) => {
+    const version = await rpc.getRpcApiVersion();
+    let results: any = {};
+    if (
+        isValidTransactionDigest(
+            query,
+            version?.major === 0 && version?.minor < 18 ? 'base64' : 'base58'
+        )
+    ) {
+        const txdata: SuiTransactionResponse =
+            await rpc.getTransactionWithEffects(query);
+        results.transaction = [
+            {
+                id: txdata.certificate.transactionDigest,
+                label: txdata.certificate.transactionDigest,
+                type: 'transaction',
+            },
+        ];
+    }
+
+    if (isValidSuiAddress(query) && !isGenesisLibAddress(query)) {
+        const data = await rpc.getObjectsOwnedByAddress(query);
```

Comment:
Instead of detecting if an address is valid this way, let's `getTransaction` with a `To` and `From` address filter and a limit of 1. This will give us an indication if an address is used or not (which will also cover cases where all assets are transferred out. Although long term we might want a dedicated RPC method to do this. we can start a thread with devx about this.

Reply:
done